### PR TITLE
[xtask] New subcommand: Books

### PR DESCRIPTION
**Pull request by syl20bnr**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Changes

Adds a new command to xtask to manage our books:

```
Run commands to manage Burn Books

Usage: xtask books <COMMAND>

Commands:
  burn         Burn Book, a.k.a. the guide, made for the Burn users
  contributor  Contributor book, made for people willing to get all the technical understanding and advices to contribute actively to the project
  help         Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

```
Burn Book, a.k.a. the guide, made for the Burn users

Usage: xtask books burn <COMMAND>

Arguments:
  <COMMAND>
          Possible values:
          - build: Build the book
          - open:  Open the book on a random port and rebuild it automatically upon changes

Options:
  -h, --help
          Print help (see a summary with '-h')
```

### CODE REVIEW

File: xtask/src/books.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    fn build(&self) {
+        run_mdbook_with_path(
+            "build",
+            Params::from([]),
+            HashMap::new(),
+            Some(self.path),
+            "mdbook should build the book successfully",
+        );
```
[the remark this answers is not in the document]
Comment (syl20bnr) [reply xtask/src/books.rs:102]: That's a good idea but I would suggest to do it in another PR and update the other methods as well. For now let's keep this consistent in the crate.

File: xtask/src/utils/process.rs [hunk 2]
```diff
@@ -29,3 +30,9 @@ pub(crate) fn run_command(command: &str, args: &[&str], command_error: &str, chi
     // Handle command child process
     handle_child_process(command, child_error);
 }
+
+/// Return a random port between 3000 and 9999
+pub(crate) fn random_port() -> u16 {
```
[the remark this answers is not in the document]
Comment (nathanielsimard) [reply xtask/src/utils/process.rs:35]: Could be an option, at some point.

[the remark this answers is not in the document]
Comment (syl20bnr) [reply xtask/src/utils/process.rs:35]: Done

**Review by Luni-4 — APPROVED** [review]
Thanks a lot!

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**